File: AtCoder_Beginner_Contest_185/C-Duodecim_Ferra.py
from math import comb


def main():
    """
    長さ L の棒の分割後の各棒の長さが全て正整数になるためには，整数長さ以外の場所を切ってはいけない．
    切る位置の候補は L-1 個あり，そこから異なる 11 箇所を選ぶ場合の数を求めれば良い．
    これは comb(L-1, 11) で求められる．
    """
    l_input = int(input())
    ans = comb(l_input-1, 11)
    return ans


if __name__ == '__main__':
    print(main())

File: AtCoder_Beginner_Contest_185/C-Duodecim_Ferra.cpp
#include <iostream>
#include <typeinfo>
using namespace std;

long long comb(int n, int r) {
  long long ncr = 1;
  for (int i = 1; i <= r; i++) {
    ncr *= n - i;  // nCr の分子
    ncr /= i;  // nCr の分母
  }
  return ncr;
}

int main() {
  int l_input;
  cin >> l_input;
  long long ans;
  ans = comb(l_input, 11);
  cout << ans << endl;

  return 0;
}
